app/src/main/java/com/openclassrooms/realestatemanager/presentation/ui/fragment/AddPropertyFragment.java

```diff
@@ -1,24 +1,198 @@
 package com.openclassrooms.realestatemanager.presentation.ui.fragment;
 
 import android.os.Bundle;
+import android.text.InputType;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
+import android.widget.ArrayAdapter;
+import android.widget.AutoCompleteTextView;
+import android.widget.CheckBox;
+import android.widget.EditText;
+import android.widget.Toast;
 
 import androidx.annotation.NonNull;
 import androidx.annotation.Nullable;
+import androidx.navigation.Navigation;
 
+import com.openclassrooms.realestatemanager.R;
 import com.openclassrooms.realestatemanager.databinding.FragmentAddPropertyBinding;
+import com.openclassrooms.realestatemanager.databinding.FragmentAddPropertyInformationLayoutBinding;
+import com.openclassrooms.realestatemanager.presentation.ui.fragment.adapter.PhotoRecyclerViewAdapter;
 import com.openclassrooms.realestatemanager.presentation.ui.main.BaseFragment;
+import com.openclassrooms.realestatemanager.presentation.utils.RecyclerViewItemClickListener;
+import com.openclassrooms.realestatemanager.presentation.utils.customView.AddPropertyInformationCustomView;
+import com.picone.core.domain.entity.Property;
+import com.picone.core.domain.entity.PropertyPhoto;
+
+import org.jetbrains.annotations.Contract;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static com.picone.core.utils.ConstantParameters.ADD_PHOTO;
+import static com.picone.core.utils.ConstantParameters.PROPERTY_TO_ADD;
 
 public class AddPropertyFragment extends BaseFragment {
 
     private FragmentAddPropertyBinding mBinding;
+    private List<PropertyPhoto> mPropertyPhotos = new ArrayList<>();
+    private List<PropertyPhoto> mPhotosToDelete = new ArrayList<>();
+    private boolean mIsPropertyValuesAreNull;
 
     @Nullable
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
         mBinding = FragmentAddPropertyBinding.inflate(getLayoutInflater());
+        mNavController = Navigation.findNavController(requireActivity(), R.id.nav_host_fragment);
+        setAppBarVisibility(false);
+        initRecyclerView();
+        setUpdateButtonIcon(false);
         return mBinding.getRoot();
     }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        mIsPropertyValuesAreNull = Objects.requireNonNull(mPropertyViewModel.getSelectedProperty.getValue()).getAddress() == null;
+
+        initView();
+        initDropDownMenu();
+        initClickListener();
+        configureOnClickRecyclerView();
+
+        mPropertyViewModel.isUpdateComplete.observe(getViewLifecycleOwner(), isUpdateComplete -> {
+            if (!isUpdateComplete) return;
```

Comment:
`selected property` is a empty Property object ?

Reply:
Not exactly but is empty if come from add button and filled if come from update button